core/test/test-helpers.js

```diff
@@ -85,6 +85,20 @@ function waitForChildListMutation(el) {
   });
 }
 
+function isElementVisible(el) {
+  if (!el) {
+    return false;
+  }
+  const styles = window.getComputedStyle(el);
+  return (
+    el.clientWidth !== 0 &&
+    el.clientHeight !== 0 &&
+    styles.opacity > 0 &&
+    styles.visibility !== 'hidden' &&
+    styles.display !== 'none'
+  );
```

Comment:
We could also check visibility [in the same way puppeteer does](https://github.com/GoogleChrome/puppeteer/blob/e0c8d46af1ce2c063778b66aa29c4c00ab8aeba5/lib/DOMWorld.js#L516).